packages/task/src/browser/task-service.ts

```diff
@@ -153,6 +153,21 @@ export class TaskService implements TaskConfigurationClient {
         return this.providedTaskConfigurations.getTasks();
     }
 
+    addRecentTasks(tasks: TaskConfiguration | TaskConfiguration[]): void {
+        if (Array.isArray(tasks)) {
+            tasks.forEach(task => this.addRecentTasks(task));
+        } else {
+            const cached = this.recentTasks.some(recent => TaskConfiguration.equals(recent, tasks));
```

Comment:
it should be:
1. remove the given task from recent tasks
2. unshift the given task to recent tasks

Reply:
fixed